SQR: Allotment processing

ahmad-elbatanouni opened the pull request.

anantakrishna commented on functions/helpers.js:
```diff
@@ -1,3 +1,13 @@
+let options = { 
+    method: 'POST',
+    url: 'https://api.sendinblue.com/v3/emailCampaigns/1/sendTest',
```
Don't you want to use https://github.com/sendinblue/APIv3-nodejs-library?

ahmad-elbatanouni replied:
done

anantakrishna commented on functions/src/sendEmail.js:
```diff
@@ -0,0 +1,37 @@
+            ref.child("status").once('value')
+            .then(snapshot => {
+                if(snapshot.exists())
+                    ref.update({status: "Given"}, err => {
```
You're updating the status of a file inside the `sendEmailOnNewAllotment` function. Not a good practice, as the method name does not reveal all the side-effects. As I mentioned in Trello, it may worth creating two functions for two actions, not to mix things together.

ahmad-elbatanouni replied:
done

anantakrishna commented on functions/src/sendEmail.js:
```diff
@@ -0,0 +1,37 @@
+
+            sendEmail(email, request, sendInBlueSecretKey);
+
+            return snapshot.ref.child('mailSent').set("Sent");
```
```suggestion
            return snapshot.ref.child('emailSent').set("true");
```

ahmad-elbatanouni replied:
done

ahmad-elbatanouni commented:
I'll read now some articles about the best practices of organizing the cloud functions files ..

Also regarding the the conflicting files, do you mind if I clone the repository and create a new pull request with the updated files ?

ahmad-elbatanouni commented:
I rebased the branch to the masted and squashed the commits as you suggested.

when I checkout to the `sqr-allotment-processing` again it gives me this message

> Switched to branch 'sqr-allotment-processing'
> Your branch and 'origin/allotment_processing' have diverged,
> and have 1 and 2 different commits each, respectively.
>   (use "git pull" to merge the remote branch into yours)

Is that okay, should I merge the remote to my local branch ?

anantakrishna commented:
No, you need to force push the branch onto the server.

On Sat, Oct 27, 2018 at 6:05 PM Ahmad ElBatanouni <notifications@github.com>
wrote:

> I rebased the branch to the masted and squashed the commits as you
> suggested.
>
> when I checkout to the sqr-allotment-processing again it gives me this
> message
>
> Switched to branch 'sqr-allotment-processing'
> Your branch and 'origin/allotment_processing' have diverged,
> and have 1 and 2 different commits each, respectively.
> (use "git pull" to merge the remote branch into yours)
>
> Is that okay, should I merge the remote to my local branch ?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/PureBhaktiArchive/audioseva/pull/4#issuecomment-433611170>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFyLrypSnRLZVWML1qOYLkCT29nyzJz2ks5upD3pgaJpZM4X4bMO>
> .
>

ahmad-elbatanouni commented:
I used a new branch as I'm having some troubles dealing with this one ... Can we work on that other one for now until I learn more about Git

anantakrishna commented:
Sure, closing this pull request.
